Static Branding Pages

ImJunny opened the pull request:
# Description
- Navigation Bar Updates
- Home Page
- About Page
- Partners Page (for Spring 2024)
- Contact page

# Test Guidelines

* Reviewed Via Preview Branches throughout this long-lived canary feature branch.
* Finalize-Home-Layouts was the "review" branch that had gone through and finalized and reviewed all these changes before merging back down to home-layouts.

ImJunny committed:
make example sections

ImJunny committed:
change spacing

ImJunny committed:
add sections

ImJunny committed:
remove nav elements

ImJunny committed:
change pics and add dividers

ImJunny committed:
add styling to page and headers

ImJunny committed:
add more styling

ImJunny committed:
change footer logo origin

ImJunny committed:
update files for layouts

ImJunny committed:
remove hardcoded headers from pages

ImJunny committed:
fix nav and header positions

ImJunny committed:
change layout names

ImJunny committed:
make onboarding layout

ImJunny committed:
fix footer

ImJunny committed:
remove protected footer and fix hero spacing

dineshUmasankar committed:
fix[hackpsh-web]: Remove package-lock.json to avoid confusion between in npm and pnpm in dev env

dineshUmasankar commented:
Removed `package-lock.json` to avoid dev container having confusion regarding what js package runner to use (npm or pnpm).

dineshUmasankar committed:
refactor[hackpsh-web]: Code Review

dineshUmasankar reviewed (commented):
* Remove unused imports to pass vercel build ✔️ 
* Bring back (routes) folder ✔️ 
* Highlight / Underline Link in footer for visual accessbility and more space / padding ✔️ 
* Replace other links with blog as the header ✔️ 
* Consider playing different lengths to make triangle in footer and better grouping for links ✔️ 
* Review Sign-In and Sign-Out Logo Spacing ✔️ 
* Fix Section bg to classname(s) ✔️ 
* Use the cn helper function ✔️ 
* Remove the CustomImg function, and replace with the natural cn function [premature optimization] ✔️ 
* Consider using the Geist font and typehero as a reference ✔️

ImJunny committed:
Made partial code review changes

ImJunny committed:
remove protected padding-bottom

dineshUmasankar commented on src/app/(main)/(routes)/(public)/page.tsx:
```diff
@@ -45,22 +36,22 @@ function HomeCountdown() {
           <div className="flex flex-col items-center">
             <p className="text-5xl">02</p>
-            <p className="text-xl">Seconds</p>
+            <p className="">Seconds</p>
```
Why have empty className tag? Just have no styling if there's no styling to the specific element at hand

ImJunny replied:
Used to have styling, then removed. Accidentally kept them after highlighting all _text-xl_ at once. This was a mistake.

dineshUmasankar commented on src/app/_components/page-assets/contact-section.tsx:
```diff
@@ -3,20 +3,20 @@ import Section from "./section";
 }: {
   title: string;
-  bg: string;
+  className: string;
```
I don't think this component should have a className or a custom title.

Is it supposed to be a presentational component with multiple sections of customizable design (title, description, primary button)? As I believe that the Section Component should be highly customizable, and the contact section should really just be the "configured" component. 

Take a look at [Shopify's React Components](https://polaris.shopify.com/components/layout-and-structure/card?example=card-with-multiple-sections). As you can see here the base component has a lot of customizability to allow a developer to extend and craft it into a purposeful specific component like a contact section. If this "configured" component is being re-used, then they should simply export the component with its configurations (no customizability), otherwise they should "reconfigure" from the Section Component for that specific page.

dineshUmasankar reviewed (commented):
- This branch still requires content in order to replace the filler text
- It also requires review of the existing comments
- Implementation of the countdown timer until the event date.

CatHub11 committed:
Merge Partners to Home layouts (#53)

* * Added gray box

* * Coded layout of partner page.

* * Adjusted "Visit" button to the left side of page.

Mmadavar committed:
remove protected padding-bottom
- Made an about section (For the static pages)
- Made reusable components from other static pages(section, contact us
   section)
- Made a officers component and helpers component
- Made a component for the About US heading for the about-page

Mmadavar committed:
Merge pull request #54 from IEEE-PSH:about-page

Made an about section (For the static pages)
Made reusable components from other static pages(section, contact us
section)
Made a officers component and helpers component
Made a component for the About US heading for the about-page


Test Guidelines
Open this branch.
type in the terminal pnpm run setup

dineshUmasankar committed:
Revert "remove protected padding-bottom"

This reverts commit b6b08638a20d56ee905af1465b2b58f5f968536a.

ImJunny committed:
make mobile footer

ImJunny committed:
merge About page manually

PSHIEEE commented on next.config.js:
```diff
@@ -1,5 +1,11 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  // //add images loaded by NextJS; currently a premature optimization
+  // images:{
+  //   loader: 'penn-state-logo',
+  //   loaderFile: ''
+
+  // },
```
Remove these comments, and when implementing the content. Ensure that the images are being locally sourced from within a public folder. 

[https://nextjs.org/docs/app/building-your-application/optimizing/static-assets](url)

dineshUmasankar committed:
Content Implementation for Home Page (#58)

* replacing placeholder content

* escaping apostrophe

* updating contact section

---------

Co-authored-by: ritz-16 <79714302+ritz-16@users.noreply.github.com>

Mmadavar committed:
Made the Partners page responsive (#59)

Mmadavar committed:
responsive changes on about page (#57)

Co-authored-by: John Nguyen <69529980+ImJunny@users.noreply.github.com>

ImJunny committed:
redesign partners

ImJunny committed:
update about page design & content

ImJunny committed:
add ieee about section

ImJunny committed:
Implement Contact Page (#62)

* update siteconfig

* add contact page to nav

* implement contact

ImJunny committed:
replace content placeholder

ImJunny committed:
change home hero title

ImJunny committed:
redesign contact page

ImJunny committed:
touch ups

ImJunny committed:
change hero text colors

ImJunny committed:
simplify footer

ImJunny committed:
update homepage images and buttons

ImJunny committed:
Home layouts countdown (#63)

* implement working countdown

* refactor

* update countdown stages

* change to accurate time

* refactor countdown

* remove "use client"

ImJunny committed:
change countdown background

ImJunny committed:
change nav height to 64px

ImJunny committed:
add asChild to buttons

ImJunny committed:
theme public pages

ImJunny committed:
make nav transparent

ImJunny committed:
smaller hero title for mobile viewports

ImJunny committed:
add member info and fix layouts

ImJunny committed:
update past events info

ImJunny committed:
update visuals

ImJunny committed:
refactor design elements

dineshUmasankar committed:
Finalize home layouts (#65)

* Upgrade Dependencies and Use @dotenvx for migrate

* Improve Image Quality in Repository

* Enable RLS on App_Contact Table

* Remove Comments - Image Optimization in NextConfig

* Main Nav Changes & About Changes & Revert CMDK

Removed Dead Imports & Revert RING CSS Change &
Added Button Link CSS Variant

* Added Location and Regisration Form in SiteConfig

Refactored Member Button & Sections

Added Inline Link Styling to Button

* fix: env vars for setup script

* Visual Adjustments on Home Page

* WIP css refactor of hero section

* WIP refactor and dep update for themes

* Footer Refactor

* Footer Touchups

* Footer Touchups and CSS Variable Adjustment

* Fixing errors to deploy preview on vercel

* Fix Vercel Deployment

* Static Components Touch Up on Home Page

* Disable no-unsafe-assignment for next-image

* Touchup Partners Page and Contact Section CTA Prop

* Brand color is primary color of shadcn theme

Touchups on rest of pages
Removed tsx components for svg and hosted them as public instead

* Removing unused imports and touch up on static pgs

* fix: ignore image src lint issues

* [WIP] Countdown Refactor

* Countdown Refactor and Color Touchups

* Generate Team Elements & Color Touch-ups

* update deps & replace apos with html entity

dineshUmasankar reviewed (approved):
Reviewed via `finalize-home-layouts` branch.
